test/unit/mdc-select/mdc-select-enhanced.test.js

```diff
@@ -0,0 +1,654 @@
+  document.body.removeChild(fixture);
+  document.body.removeChild(menuSurface);
+});
+
+test('menu surface opened event causes selected element to be focused', () => {
+  const hasMockFoundation = false;
+  const hasMockMenu = false;
+  const hasOutline = false;
+  const hasLabel = true;
+  const {fixture, component, menuSurface} = setupTest(hasOutline, hasLabel, hasMockFoundation, hasMockMenu);
+  document.body.appendChild(fixture);
+  component.selectedIndex = 1;
+  const event = document.createEvent('Event');
+  event.initEvent(MDCMenuSurfaceFoundation.strings.OPENED_EVENT, false, true);
+  menuSurface.dispatchEvent(event);
+
+  assert.equal(document.activeElement, menuSurface.querySelector('.mdc-list-item--selected'));
+  assert.equal(component.selectedIndex, 1);
+  document.body.removeChild(fixture);
+  document.body.removeChild(menuSurface);
+});
+
+test('menu surface closed event calls foundation blur if selected-text is focused', () => {
+  const {fixture, menuSurface, mockFoundation} = setupWithMockFoundation();
+  document.body.appendChild(fixture);
+  fixture.querySelector('.mdc-select__selected-text').focus();
+  const event = document.createEvent('Event');
+  event.initEvent(MDCMenuSurfaceFoundation.strings.CLOSED_EVENT, false, true);
+  menuSurface.dispatchEvent(event);
+
+  td.verify(mockFoundation.handleBlur(), {times: 0});
+  document.body.removeChild(fixture);
+});
+
+test('menu surface closed event does not call foundation blur if selected-text is not focused', () => {
+  const hasMockFoundation = true;
+  const hasMockMenu = false;
+  const hasOutline = false;
+  const hasLabel = true;
+  const {fixture, menuSurface, mockFoundation} = setupTest(hasOutline, hasLabel, hasMockFoundation, hasMockMenu);
+  document.body.appendChild(fixture);
+  const event = document.createEvent('Event');
+  event.initEvent(MDCMenuSurfaceFoundation.strings.CLOSED_EVENT, false, true);
+  menuSurface.dispatchEvent(event);
+
+  td.verify(mockFoundation.handleBlur(), {times: 1});
+  document.body.removeChild(fixture);
+  document.body.removeChild(menuSurface);
+});
+
+test('keydown event is added to selected-text when initialized', () => {
+  const {fixture, mockFoundation} = setupWithMockFoundation();
+  document.body.appendChild(fixture);
+  domEvents.emit(fixture.querySelector('.mdc-select__selected-text'), 'keydown');
+  td.verify(mockFoundation.handleKeydown(td.matchers.anything()), {times: 1});
+  document.body.removeChild(fixture);
+});
+
+test('keydown event is removed from selected-text when destroyed', () => {
+  const {fixture, mockFoundation, component} = setupWithMockFoundation();
+  document.body.appendChild(fixture);
+  component.destroy();
+  domEvents.emit(fixture.querySelector('.mdc-select__selected-text'), 'keydown');
+  td.verify(mockFoundation.handleKeydown(td.matchers.anything()), {times: 0});
+  document.body.removeChild(fixture);
+});
+
+test('open menu when menu is opened does not set ', () => {
```

Comment:
I think this test is unfinished... it looks like the description was cut off and the body looks similar to the keydown event deregistration test.